### PR TITLE
Use default biblatex `in` macro.

### PR DESCRIPTION
**Pull request by DavidLuptak**
"In" macro defined in `iso.bbx` seems to be redundant since "in:" macro defined in `biblatex.def` exists.

### CODE REVIEW

**Comment by moewew** [comment]
Again, you will have to wait for @michal-h21 to confirm this, but my understanding was that due to the (frankly ridiculous) requirements for multilingual bibliographies we need `\mainsstring` instead of `\bibstring` here, so we can't use the default, but I'm all for making it more like the default


```
\renewbibmacro*{in:}{%
  \printtext{%
    \mainsstring{in}\intitlepunct}}
```

**Comment by michal-h21** [comment]
@moewew in this case it should be exactly string "In:". I ran `grep "{in}" *.lbx` in biblatex's lbx dir and it seems that it isn't translated in any lbx file. So it doesn't really make a difference whether we use `\bibstring` and `\mainsstring` in this case?

@DavidLuptak thanks as usual :)

**Comment by moewew** [comment]
Well, in `greek.lbx` we have `in = {{στο}{στο}},`, so `in` is translated, it just happens that it often comes out as "in".

**Comment by michal-h21** [comment]
Oh, maybe we should then use `In:` directly, without use of translation
strings? Other option is to use `\mainsstring` as you proposed, but it
seems that it shouldn't be translated at all.

**Comment by moewew** [comment]
I can't believe that this string is *never* translated. Why would it say "in" in a, say, Russian document with all the rest being in Cyrillic letters?

I can see a point in writing the "in" in the document main language and not the language of the entry, but that's about it.

**Comment by michal-h21** [comment]
@moewew that's true, it would look strange in non-Latin languages. In English version of norm is stated only that `In:` separator should be used, but it really doesn't mean that it should be used in all languages.  On the other hand, I've seen `In:`  used in Czech books and articles, not only those which used ISO 690, but also other styles, so it seems that it is our local custom. Will use the main language.

**Comment by moewew** [comment]
Is "in" not the proper word for it then in Czech? `czech.lbx` has a few comments after the `in` bibstring. 

```
  in               = {{in}{in}},% Pouziti napriklad u sborniku nebo
  inseries         = {{in}{in}},% casopisu, zni to lip new "v".
  % V pripade potreby pouzit lokalni lbx soubor
```

**Comment by DavidLuptak** [comment]
@moewew @michal-h21 Right! I have realized my misunderstanding of the norm itself in this issue case. In the section dealing with "In:" there is also a note that says

> In other languages, a connector other than “In” may be preferred..

with the consecutive example in Finnish. That was the impulse for me to thinking about this issue, because somehow I had imagined that when the resource is in another language than the entire document, then also this "in" string should be in that language, not the main language of the document. At this point I was wrong!

You are also right about the "in" localisation string, but according to the ISO 690 norm there is another case with this -- transliteration; which means that every non-Roman element should be transliterated or romanized. Hence mentioned examples of Greek, Russian, etc. doesn't apply to help, but have a look at the Finnish or Spanish "in" localisation strings. And at this point it makes sense to me to use `\mainsstring` macro instead of default `\bibstring` one.

@moewew a few comments in `czech.lbx` says just that "in" sounds better than czech translation of "in" ("v" in czech) and also that there could be used locally modified lbx file to overwrite the original to use literal translation.

PS1: shouldn't be `\mainsstring`-ed also another conjunctions like "and" (e.g. multiple authors)?
PS2: how we can deal with transliteration in biblatex? Just at the level of creating a .bib file?
PS3: an interesting point is with volume entry field, because the norm states that

> The edition should be specified in the terms, including symbols, used in the preferred source.

Which means that if the document is written in Czech and bib-entry contains `edition = {New enlarged edition}` in English due to some English resource, that's ok. But what should be printed out there if more common `edition = 42` appears? "42. vyd." due to Czech document or "42th ed." due to English resource ((and we suppose that creator of bib file kept in mind the recommendation of biblatex documentation that it should be an integer put in this field) and (we can suppose also that in the preferred source of data is not just an integer written, but something like "42th Edition"))? Total insanity, isn't it?

BTW thank you both for discussion and correction!

**Comment by moewew** [comment]
> according to the ISO 690 norm there is another case with this -- transliteration; which means that every non-Roman element should be transliterated or romanized. Hence mentioned examples of Greek, Russian, etc. doesn't apply to help

I'm not quite sure what transliteration has to do with that. Does that mean if I'm a Russian, writing a Russian document citing a Russian work I need to transliterate the reference to said Russian work? I would have expected that it would be uncontroversial that in that case I would use the Russian version of the "in" bibstring.

----

In all other aspects, transliteration is still something that needs better handling in `biblatex`, there were once plans for a multiscript version that would support transliterations natively, but it is now dormant.

-----

I'm not sure what should be using `mainstring` what what should be in `bibstring`. The problem is that `bibstring` is the standard and the more `mainstring` we need, the more code we need to duplicate. Maybe it is possible to go the opposite way: Let `bibstring` default to the main document language and then have a special `...string` that takes the `langid` of the current entry.

**Comment by DavidLuptak** [comment]
I understand the transliteration in a way that in Russia the norm itself might not be even available (e.i. Russian version of the norm), hence the original (English) version does not adjust this specific use. But of course, yes, it would be natural to use the respective localization string in this case.

BTW, how/where can I find all of the translations of the norm? Because, I would say, translations of the norm are typically ensured by national offices and that is impossible to skim all of them just to get list of translation, isn't it?

Redesign you proposed sounds good, I will be thinking about it as well.

**Comment by michal-h21** [comment]
>  a few comments in czech.lbx says just that "in" sounds better than czech translation of "in" ("v" in czech) and also that there could be used locally modified lbx file to overwrite the original to use literal translation.

(it is in fact my comment, seems quite stupid now :D). the thing is that in ČSN ISO 690 literal "In:" is requested, not "V:". I also never seen "V:" in any Czech bibliography, regardless of used citation style. In is always used.

But anyway, it seems to me that it would be correct to fix `czech.lbx` and include actual translation of the term, and put the modified version to `czech-iso.lbx`.

> PS1: shouldn't be \mainsstring-ed also another conjunctions like "and" (e.g. multiple authors)?

yes, it should. I haven't take a look at your other patch yet, sorry, it is quite busy week :(

> PS2: how we can deal with transliteration in biblatex? Just at the level of creating a .bib file?

at the moment I think author must provide it in the bib file. I know there are some translation tables between various languages for Unicode, but I have no idea how difficult it might be to support it in Biber.

> PS3: an interesting point is with volume entry field, because the norm states that

it should be in document language, as other terms specified in the document (so journal volume, number, etc.). ideally one should include exactly what was specified in the document, but that would be insane, of course. It just seems that librarians designed it. Take a look at MARC 21, if you want to see something really insane.

**Comment by DavidLuptak** [comment]
> But anyway, it seems to me that it would be correct to fix `czech.lbx` and include actual translation of the term, and put the modified version to `czech-iso.lbx`.

I don't think there is a need for correction, because as you said above, it seems to be a kind of bibliography custom to use "In" instead of literal translation.
Additionally, string "V: " (Czech translation of "At: ") is better suited for use in the case of chapter 13 of the norm -- Location. E.i. location of the repository that holds limited number of copies of a reference; or location of library or repository that holds graphic work.

Well, sorry for the typo in PS3, obviously there should be edition entry field, not volume one (I have corrected it). Never mind, what we can imagine by this statement in chapter 10 of the norm -- Numeration and pagination?

> The components of an item should be cited in the terms used in that information resource,..

Should be there volume, number, pages, etc. in language of resource, then?

---
To sum up, I have tried to distinguish which terms should be printed in document language and which one in the language of resource. I have ended up with this:
**document language:**
* medium designation

**Comment by DavidLuptak** [comment]
> But anyway, it seems to me that it would be correct to fix czech.lbx and include actual translation of the term, and put the modified version to czech-iso.lbx.

I don't think there is a need for correction, because as you said above, it seems to be a kind of bibliography custom to use "In" instead of literal translation.
Additionally, string "V: " (Czech translation of "At: ") is better suited for use in the case of chapter 13 of the norm -- Location. E.i. location of the repository that holds limited number of copies of a reference; or location of library or repository that holds graphic work.

Well, sorry for the typo in PS3, obviously there should be edition entry field, not volume one (I have corrected it). Never mind, what we can imagine by this statement in chapter 10 of the norm -- Numeration and pagination?

> The components of an item should be cited in the terms used in that information resource,..

Should be there volume, number, pages, etc. in language of resource, then?

To sum up, I have tried to distinguish which terms should be printed in document language and which one in the language of resource. I have ended up with this:
**in document language:**
* medium designation (chapter 7)
* type of medium (15.2.2)
* In: (6.4)
* At: (13)
* Date of citation (15.2.5)
* Availability (15.2.6)
* System requirements (15.2.3)
* Creator's role (5.4.5, 15.5.2, 15.7.1)
* Cartographic materials' info (15.5.5--7)
* "and", "andmore" and "andothers" strings
* notes, additional information, explanations (usually placed in brackets)

**in resource language:**
* Creator (5)
* Title (6)
* Edition (8)
* Production info (9)
* Numbering and pagination? (10)
* Series title and number (11)
* Additional general information (14)

Well, this list is not complete as well as there are many cases which can be handle just at the level of bib file.

**Comment by moewew** [comment]
So it seems that lots is done in the document language.

Among those fields in resource language you list "creator" and "title", but you explicitly list the creator''s role as well as other strings that could appear in the context as doc language. From that I infer that we don't need ton bother about those two fields (they don't come with additional localisation strings).

The only interesting thing is `edition`, series title and number (esp. the number), and to a certain extent - where it uses localisation - production info (what does that entail?).

What is meant by "numbering and pagination"? I would find it odd to say "S. 15" for a German document, but "p. 16" for an English one. That makes one just seem very fickle.

**Comment by DavidLuptak** [comment]
Yes, maybe I started to missing a point what should be there in resource language, because "creator" and "title" should be given in resource language at the level of bib file, there are no localization strings. Just metadata that could be added to them, such as creator's role, medium type, etc. set at the level of bib file, but these can be parsed and localized, not print as they are in the bib file.

Production info means place of production, publisher and date, in general. This apply to entry fields "publisher" and "location" at the level of bib file, no localization strings again.

Numbering and pagination means exactly what you wrote above. So, it covers volume, (part) number, range of pages, let's say also chapters, etc. Note that number of pages when citing a whole document is just optional and should be stated in brackets (this is missing in `biblatex-iso690` yet).

> I would find it odd to say "S. 15" for a German document, but "p. 16" for an English one.

But that's defined in german.lbx file exactly this way.

Series title seems to be easy to handle since there is "series" entry field. But number remains interesting.

**Comment by moewew** [comment]
>>I would find it odd to say "S. 15" for a German document, but "p. 16" for an English one.

> But that's defined in german.lbx file exactly this way.

Blast! I meant to say '"S. 15" for a German *source*, but "p. 16" for an English one within the same document'.

> Production info means place of production, publisher and date, in general. This apply to entry fields "publisher" and "location" at the level of bib file, no localization strings again.

But the date field uses localisation in `biblatex`.

> But number remains interesting.
We do have the `number` field for series...

**Comment by DavidLuptak** [comment]
Definitely, also as @michal-h21 said, edition and similar integer terms should be in document language.

> But the date field uses localisation in biblatex.

Yes, I forgot to mention the date, but it is in ISO 8601 which prints the date in the same format for all languages, isn't it? (I am really not sure about this)

**Comment by moewew** [comment]
When you say "document language" do you mean source language or (main) language of the document I'm currently writing? 

ISO 8601 would be the obvious choice, but I think I have read somewhere that ISO 690 doesn't actually want to prescribe a date format and uses different formats throughout the examples to avoid taking sides. But @michal-h21 will know more about this.

**Comment by DavidLuptak** [comment]
> When you say "document language" do you mean source language or (main) language of the document I'm currently writing? 

Main language of the document -- language in which I am writing it.

> I think I have read somewhere that ISO 690 doesn't actually want to prescribe a date format and uses different formats throughout the examples to avoid taking sides.

This is literally stated in the norm:

> If a more precise date is required, it may be shown:
a) in numerals, in the order year, month, day and, if necessary, hour, minute, second (see ISO 8601);
EXAMPLE 1    1984-02-10, 13:43:11.
b) with the month spelled out, either in full or abbreviated, and with day, month, year and time given in the order used in the preferred sources (see 4.1);
EXAMPLE 2    10 Feb. 1984.
c) if appropriate, as shown in the preferred source, e.g. the name of a season or a university term.

And it sounds like you are free to choose an option you want.

**Comment by michal-h21** [comment]
On Fri, Apr 8, 2016 at 7:36 PM, Dávid Lupták <notifications@github.com>
wrote:

> But anyway, it seems to me that it would be correct to fix czech.lbx and
> include actual translation of the term, and put the modified version to
> czech-iso.lbx.
>
> I don't think there is a need for correction, because as you said above,
> it seems to be a kind of bibliography custom to use "In" instead of literal
> translation.
>
OK

> Additionally, string "V: " (Czech translation of "At: ") is better suited
> for use in the case of chapter 13 of the norm -- Location. E.i. location of
> the repository that holds limited number of copies of a reference; or
> location of library or repository that holds graphic work.
>
The funny thing is that Czech translation states literal "At:" in chapter
13 as well... I don't know whether this is our local bibliographic custom,
this information isn't included in references too much often. Personally, I
think this is really weird.

Anyway, this information isn't included in the style at the moment, I think
it should be recorded in "library" Biblatex field.

> Well, sorry for the typo in PS3, obviously there should be edition entry
> field, not volume one (I have corrected it). Never mind, what we can
> imagine by this statement in chapter 10 of the norm -- Numeration and
> pagination?
>
> The components of an item should be cited in the terms used in that
> information resource,..
>
> Should be there volume, number, pages, etc. in language of resource, then?
>

Volume and number are in language of resource, the pages aren't. Maybe it
is because there are usually only numbers? It is definitely weird and
possibly translation error, but in all examples in the Czech translation,
pages are translated.

**Comment by michal-h21** [comment]
Nice list, thank you. It seems quite accurate, we should add support for
some missing items.

> To sum up, I have tried to distinguish which terms should be printed in
> document language and which one in the language of resource. I have ended
> up with this:
> *in document language:*
>
>    - medium designation (chapter 7)
>    - type of medium (15.2.2)
>    - In: (6.4)
>    - At: (13)
>
> (it isn't used yet)

>
>    - Date of citation (15.2.5)
>    - Availability (15.2.6)
>
>
>    - System requirements (15.2.3)
>
> (it isn't used as well. it should be maybe included only in note? I don't
think there is a corresponding Biblatex field)

>
>    - Creator's role (5.4.5, 15.5.2, 15.7.1)
>    - Cartographic materials' info (15.5.5--7)
>    - "and", "andmore" and "andothers" strings
>    - notes, additional information, explanations (usually placed in
>    brackets)
>
> (they aren't used as well.)

I think pagination should be in the document language as well, unless it is
just an translation error. Maybe it should be used only in Czech? (It would
be really messy)

>
>
> *in resource language:*
>
>    - Creator (5)
>    - Title (6)
>    - Edition (8)
>    - Production info (9)
>    - Numbering and pagination? (10)
>    - Series title and number (11)
>    - Additional general information (14)
>
> Well, this list is not complete as well as there are many cases which can
> be handle just at the level of bib file.
>
Yes, some things should be handled at the level of bib file, for example in
edition, bibstring is printed only if the field is only number, when it
contains some additional information, it is printed as it is. Maybe similar
mechanism could be used also for volume, issue and other fields.

**Comment by michal-h21** [comment]
> The only interesting thing is edition, series title and number (esp. the
> number), and to a certain extent - where it uses localisation - production
> info (what does that entail?).
>
There is also quite extensive stuff in regards of online documents: type of
source (online, cd-rom, etc), date of citation, information about
availability.

**Comment by michal-h21** [comment]
>
> Numbering and pagination means exactly what you wrote above. So, it covers
> volume, (part) number, range of pages, let's say also chapters, etc. Note
> that number of pages when citing a whole document is just optional and
> should be stated in brackets (this is missing in biblatex-iso690 yet).
>
Optionality of number of pages seems like change in the current version, it
used to be required for books. It is quite strange to put them in square
brackets, but it probably follows logic of putting all additional
information to the square brackets.

**Comment by michal-h21** [comment]
>
> Yes, I forgot to mention the date, but it is in ISO 8601 which prints the
> date in the same format for all languages, isn't it? (I am really not sure
> about this)
>

Yes, date should be language independent, in form YYYY-MM-DD. It is used in
date of citation, year suffices as date of publication.

**Comment by DavidLuptak** [comment]
> The funny thing is that Czech translation states literal "At:" in chapter 13 as well... I don't know whether this is our local bibliographic custom, this information isn't included in references too much often. Personally, I think this is really weird.

Not exactly, let have a look at section 15.7.2 where very similar example is stated. "At:" is translated there. And funny thing follows, data in square brackets are translated into Czech in both examples and in both cases any of that two words are not the same :D

> Anyway, this information isn't included in the style at the moment, I think it should be recorded in "library" Biblatex field.

Yes, could be the most appropriate place for this.

> Volume and number are in language of resource, the pages aren't. Maybe it is because there are usually only numbers?

@michal-h21 But what about this discussion according to document VS resource language

> I meant to say '"S. 15" for a German source, but "p. 16" for an English one within the same document'.

Isn't it a violation against consistency of creating references mentioned as basic principle in the norm?? There is stated:

> A uniform style, format and punctuation scheme shall be used for all references in a document,
regardless of the particular style guide being used.

I am really not sure about what this exactly means.

**Comment by michal-h21** [comment]
@DavidLuptak I think that the norm tries to say that you can choose whether form you like, you should just stick with it in all your references. And then it fails to manage the consistency in the norm itself, probably to give us a warning how we could confuse the readers without a consistency. It does really good job in this task :)
